code/modules/projectiles/guns/energy/energy_gun.dm

```diff
@@ -11,6 +11,33 @@
 	flight_y_offset = 10
 	dual_wield_spread = 60
 
+/obj/item/gun/energy/e_gun/cyborg
+	can_charge = FALSE
+	name = "cyborg energy gun"
+	desc = "An integrated energy gun that draws from a cyborg's power cell. This weapon contains a limiter that keeps cyborgs from firing it on a lethal setting, but an emagged cyborg could probably bypass it." //hint hint nudge nudge
+	use_cyborg_cell = TRUE
+
+/obj/item/gun/energy/e_gun/cyborg/emp_act()
+	return
+
+//the following secborg energy gun code is mostly copied from Fulpstation's secborg patch PR (thanks for letting us use this, Surrealaser!)
+/obj/item/gun/energy/e_gun/cyborg/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
+	if(!check_limiter())
+		return
+	return ..()
+
+/obj/item/gun/energy/e_gun/cyborg/proc/check_limiter()
+	var/mob/living/silicon/robot/R = loc
+	if(!R || !iscyborg(R))
+		return FALSE
+	var/obj/item/ammo_casing/energy/shot = ammo_type[select]
+	if(shot.harmful && !R.emagged) //If we're emagged, we don't care about whether we're firing lethal shots or not
+		playsound(loc, 'sound/machines/buzz-two.ogg', get_clamped_volume(), TRUE, -1) //buzzkill alert
+		to_chat(loc,"<span class='warning'>[src]'s limiter prevents you from firing it on a lethal setting!</span>")
+		return FALSE
+	return TRUE
```

Comment:
Instead of checking every shot you can guarantee it starts non-harmful and prevent it from changing while not emagged.

Reply:
Yeah, I'll see if I can copy some of this over to attack_self() (I still want to keep this here for emagged secborgs who somehow become un-emagged, though).

Reply:
It would be cleaner to do that change when the un-emagging happens instead of here.